File: 2888-minimum-index-of-a-valid-split/minimum-index-of-a-valid-split.py
class Solution:
    def minimumIndex(self, nums: List[int]) -> int:
        candidate, votes, n = -1, 0, len(nums)
        for i in range(n):
            if (votes == 0):
                candidate = nums[i]
                votes = 1
            else:
                if (nums[i] == candidate):
                    votes += 1
                else:
                    votes -= 1

        votes = 0
        for j in range(n):
            if nums[j] == candidate:
                votes += 1
                if 2*votes > j+1:
                    return j if 2*nums[j+1:].count(candidate) > n-j-1 else -1
        return -1

File: 2888-minimum-index-of-a-valid-split/minimum-index-of-a-valid-split.cpp
class Solution {
public:
    int minimumIndex(vector<int>& nums) {
        int votes = 0, candidate = -1, n = nums.size();
        for (int num: nums) {
            if (!votes) {
                candidate = num;
                votes++;
            } else {
                if (candidate == num) votes++;
                else votes--;
            }
        }
        votes = 0;
        for (int j=0; j<n; j++) {
            if (nums[j] == candidate) {
                votes++;
                if (2*votes > j+1) {
                    if (2*count(nums.begin()+j+1, nums.end(), candidate) > n-j-1) return j;
                    else return -1;
                }
            }
        }
        return -1;
    }
};